app/src/test/java/edu/wpi/first/shuffleboard/app/plugin/PluginLoaderTest.java

```diff
@@ -0,0 +1,30 @@
+package edu.wpi.first.shuffleboard.app.plugin;
+
+import edu.wpi.first.shuffleboard.api.plugin.Plugin;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+public class PluginLoaderTest {
+
+  public static class MockPlugin extends Plugin {
+
+    public MockPlugin() {
+      super("test", "Mock Plugin", "0.0.0", "A plugin for testing");
+    }
+
+  }
+
+  @Test
+  public void testLoadClass() {
+    PluginLoader loader = new PluginLoader();
+    boolean loaded = loader.loadPluginClass(MockPlugin.class);
```

Comment:
As soon as you try doing this with more complicated classes that actually have plugin logic you are going to end up adding stuff to the `Widgets` singleton and other singletons which will affect other tests.